### PR TITLE
Feature/add gist option

### PR DESCRIPTION
**Pull request by yotamolenik**
Payload_types:
  - Refactor the parsing to use OOP
  - Add an option for md file output

### CODE REVIEW

File: mobileconfig/mobileconfig.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    def __repr__(self):
+        inner_payload_content = self.payload_content
+        domain = inner_payload_content[0].get('DefaultsDomainName')
+        data_keys = ', '.join(inner_payload_content[0].get('DefaultsData').keys())
+        description = f'{domain}: {data_keys}'
+        return description
+
+
+class CoreCapture(PayloadContent):
+
```
Comment (doronz88) [inline mobileconfig/mobileconfig.py:64]: Remove these lines

File: mobileconfig/__main__.py [hunk 2]
```diff
@@ -2,9 +2,12 @@
 import plistlib
 from pathlib import Path
 from typing import Mapping
+import mdutils
 
 import click
-from humanfriendly.tables import format_smart_table
+
+# im not yet sure why, but pycharm thinks this import is wrong. in fact it works perfectly fine on my terminal
```
Comment (doronz88) [inline mobileconfig/__main__.py]: Are you?

File: mobileconfig/__main__.py [hunk 3]
```diff
@@ -2,9 +2,12 @@
 import plistlib
 from pathlib import Path
 from typing import Mapping
+import mdutils
```
Comment (doronz88) [inline mobileconfig/__main__.py:5]: Reorder your imports using pycharm. Third party ones should be grouped together

File: mobileconfig/__main__.py [hunk 4]
```diff
@@ -6,8 +6,7 @@
 
 import click
 
-# im not yet sure why, but pycharm thinks this import is wrong. in fact it works perfectly fine on my terminal
```
Comment (doronz88) [inline mobileconfig/__main__.py]: Dont include this in the first commit in the first place if this is removed

**Review by doronz88 — CHANGES_REQUESTED** [review]
Fix comments and add a link into the gist page from the main readme

File: mobileconfig/__main__.py (hunk 2 above)
Comment (yotamolenik) [reply mobileconfig/__main__.py]: during the time of the 1st commit, yes. but i can remove it too now

File: requirements.txt [hunk 5]
```diff
@@ -1,2 +1,3 @@
 humanfriendly
```
[the remark this answers is not in the document]
Comment (yotamolenik) [reply requirements.txt]: nope, now removed

File: mobileconfig/mobileconfig.py (hunk 1 above)
Comment (yotamolenik) [reply mobileconfig/mobileconfig.py:64]: done

File: mobileconfig/__main__.py (hunk 3 above)
Comment (yotamolenik) [reply mobileconfig/__main__.py:5]: done

File: mobileconfig/mobileconfig.py [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+            'com.apple.defaults.managed': ManagedPayload,
+            'com.apple.corecapture.configure': CoreCapture,
+            'com.apple.ManagedClient.preferences': ManagedClient
+        }[plist['PayloadType']](plist)
+
+    def __init__(self, plist: Mapping):
+        self._plist = plist
+
+    @property
+    def payload_content(self) -> 'PayloadContent':
```
Comment (doronz88) [inline mobileconfig/mobileconfig.py:21]: this doesn't return the `PayloadContent`, just the RAW plist. return the correct object and refactor accordingly

File: README.md [hunk 7]
```diff
@@ -14,7 +14,11 @@ Options:
...
   consents       Print all user consents
-  extract        Extract .plist into given directory
-  payload-types  Print all PayloadType's (with an optional .csv output)
+  extract        Extract the plists into given directory
+  payload-types  print a summary of PayloadTypes to either a csv or md file
 ```
 
+The _extract_ command also gives all values for the keys inside payload-types
+
+for your convenience, a link to the gist of all current profiles:
```
Comment (doronz88) [inline README.md:23]: `For`

**Review by doronz88 — CHANGES_REQUESTED** [review]
It's almost there

File: mobileconfig/mobileconfig.py (hunk 6 above)
Comment (yotamolenik) [reply mobileconfig/mobileconfig.py:21]: oops, right

File: README.md (hunk 7 above)
Comment (yotamolenik) [reply README.md:23]: done

File: mobileconfig/mobileconfig.py [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+    def payload_content(self) -> Mapping:
+        return self._plist['PayloadContent']
+
+    @property
+    def payload_type(self) -> str:
+        return self._plist['PayloadType']
+
+
+class LoggingPayload(PayloadContent):
+
```
Comment (doronz88) [inline mobileconfig/mobileconfig.py:30]: remove this line

**Review by doronz88 — CHANGES_REQUESTED** [review]
another extra line

File: mobileconfig/mobileconfig.py (hunk 8 above)
Comment (yotamolenik) [reply mobileconfig/mobileconfig.py]: done

File: mobileconfig/mobileconfig.py [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+    @property
+    def payload_content(self) -> Mapping:
+        return self._plist['PayloadContent']
+
+    @property
+    def payload_type(self) -> str:
+        return self._plist['PayloadType']
+
+
+class LoggingPayload(PayloadContent):ֿ
```
Comment (doronz88) [inline mobileconfig/mobileconfig.py:29]: Remove the character at the end of this line

**Review by doronz88 — CHANGES_REQUESTED** [review]
This code cannot work

File: mobileconfig/mobileconfig.py (hunk 9 above)
Comment (yotamolenik) [reply mobileconfig/mobileconfig.py]: done

File: mobileconfig/__main__.py [hunk 10]
```diff
@@ -2,9 +2,11 @@
 import plistlib
 from pathlib import Path
 from typing import Mapping
+import mdutils
```
Comment (doronz88) [inline mobileconfig/__main__.py:5]: reorder imports

**Review by doronz88 — CHANGES_REQUESTED** [review]

**Review by doronz88 — APPROVED** [review]

**Pull request merged**